src/app/children/child-details/child-details.component.ts

```diff
@@ -154,10 +155,14 @@ export class ChildDetailsComponent implements OnInit {
           this.router.navigate(['/child', this.child.getId()]);
           this.creatingNew = false;
         }
+        this.alertService.addInfo('Saving Succesfull');
         this.switchEdit();
       })
       .catch((err) => this.alertService.addDanger('Could not save Child "' + this.child.name + '": ' + err));
+  } else {
+    this.alertService.addDanger('Form ist not valid!!');
```

Comment:
At the moment, it is unclear (or even impossible?) for the user to see what he/she should do to make the form valid.

We have to mark the required form fields visibly as "required" and it would be much easier for users if we highlight the individual invalid field(s) with a specific error message, as in the [live example](https://angular.io/generated/live-examples/form-validation/stackblitz.html) from the [Angular Docs](https://angular.io/guide/form-validation)), for example.

Reply:
Good point, I'll look further into that :)